wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth1/OAuthAuthenticator.java

```diff
@@ -0,0 +1,372 @@
+     * @return the authorization header value, or {@code null} if token was not found for given user id.
+     * @throws OAuthAuthenticationException
+     *         if authentication failed.
+     */
+    String computeAuthorizationHeader(@NotNull final String userId,
+                                      @NotNull final String requestMethod,
+                                      @NotNull final String requestUrl) throws OAuthAuthenticationException {
+        final OAuthCredentialsResponse credentials = new OAuthCredentialsResponse();
+        OAuthToken oauthToken = getToken(userId);
+        credentials.token = oauthToken != null ? oauthToken.getToken() : null;
+        if (credentials.token != null) {
+            return computeAuthorizationHeader(requestMethod, requestUrl, credentials.token, credentials.tokenSecret);
+        }
+        return null;
+    }
+
+    private OAuthToken getToken(final String userId) {
+        OAuthCredentialsResponse credentials;
+        credentialsStoreLock.lock();
+        try {
+            credentials = credentialsStore.get(userId);
+        } finally {
+            credentialsStoreLock.unlock();
+        }
+        if (credentials != null) {
+            return newDto(OAuthToken.class).withToken(credentials.token).withScope(credentials.tokenSecret);
+        }
+        return null;
+    }
+
+    /**
+     * Compute the Authorization header to sign the OAuth 1 request.
+     *
+     * @param requestMethod
+     *         the HTTP request method.
+     * @param requestUrl
+     *         the HTTP request url with encoded query parameters.
+     * @param token
+     *         the token.
+     * @param tokenSecret
+     *         the secret token.
+     * @return the authorization header value, or {@code null}.
+     */
+    private String computeAuthorizationHeader(@NotNull final String requestMethod,
+                                              @NotNull final String requestUrl,
+                                              @NotNull final String token,
+                                              @NotNull final String tokenSecret) throws OAuthAuthenticationException {
+        OAuthParameters oauthParameters;
+        try {
+            oauthParameters = new OAuthParameters();
+            oauthParameters.consumerKey = clientId;
+            oauthParameters.signer = clientSecret == null ? getOAuthRsaSigner() : getOAuthHmacSigner(clientSecret, tokenSecret);
+            oauthParameters.token = token;
+            oauthParameters.version = "1.0";
+
+            oauthParameters.computeNonce();
+            oauthParameters.computeTimestamp();
+
+            oauthParameters.computeSignature(requestMethod, new GenericUrl(requestUrl));
+        } catch (GeneralSecurityException e) {
+            throw new OAuthAuthenticationException(e);
+        }
+
+        return oauthParameters.getAuthorizationHeader();
+    }
+
+    private String getUserFromStateParameter(final String state) {
```

Comment:
`getUserFromStateParameter`, `getSignatureMethodFromStateParameter`, `getRequestTypeFromStateParameter` methods are very similar. Why we can not create method `getParameterFromState(String state, String parameterName)`?